File: 1-D Dynamic Programming/Min Cost Climbing Stairs/MinClimb.java
import java.util.Scanner;

class Solution {
  public int minCostClimbingStairs(int[] cost) {
    int n = cost.length;
    for (int i = 2; i < n; i++) {
      cost[i] += Math.min(cost[i - 1], cost[i - 2]);
    }
    return Math.min(cost[n - 1], cost[n - 2]);
  }
}

public class MinClimb {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    Solution obj = new Solution();
    int tc = sc.nextInt();
    while (tc-- > 0) {
      int n = sc.nextInt();
      int cost[] = new int[n];
      for (int i = 0; i < n; i++) {
        cost[i] = sc.nextInt();
      }
      System.out.println(obj.minCostClimbingStairs(cost));
    }
    sc.close();
  }
}

File: 1-D Dynamic Programming/Min Cost Climbing Stairs/MinClimb.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        for(int i = 2; i < n; i++) {
            cost[i] += min(cost[i - 1], cost[i -2]);
        }
        return min(cost[n - 1], cost[n - 2]);
    }
};

int main() {
  Solution obj;
  int tc;
  cin >> tc;
  while (tc-- > 0) {
    int n, amount;
    cin >> n;
    vector<int> cost(n);
    for (int i = 0; i < n; i++) {
      cin >> cost[i];
    }
    cout << (Solution().minCostClimbingStairs(cost)) << endl;
  }
  return 0;
}